### ISSUE TITLE
Inconsistent behavior of batch mode and interactive mode

### ISSUE THREAD

**Issue by aseemr** [FStarLang/FStar#854]
On this code:

```
module Test

let x : Type * int = (unit, 5)
unfold let first (#a:Type) (#b:Type) (x: (a * b)) : Tot a =
  match x with
  | (a, _) -> a
let z : first x = ()
```

The behavior of batch mode and interactive mode differs:

```
$ fstar.exe Test.fst
Unexpected error; please file a bug report, ideally with a minimized version of the source program that triggered the error.
Failure("(.\\Test.fst(3,8-3,12)) Impossible: Forced tk not present on Type")
```

vs

```
$ fstar.exe Test.fst --in
#push 1 0
module Test

let x : Type * int = (unit, 5)
unfold let first (#a:Type) (#b:Type) (x: (a * b)) : Tot a =
  match x with
  | (a, _) -> a
let z : first x = ()
#end #done-ok #done-nok

#done-ok
```

The blowup of the batch mode invocation is also an issue.

**Comment by aseemr** [comment]
Removing the `unfold` keyword makes it go through even in the batch mode (on the `c_relational-ci_r3` branch with the fix for #850).

**Comment by mtzguido** [comment]
Hey Aseem. This seems to work now in master after the removal of the `tk` field.

Want to check if some variation still blows up, or should we close this?
